.github/workflows/release.yaml

```diff
@@ -333,3 +333,43 @@ jobs:
       base64-subjects: "${{ needs.combine_hashes.outputs.hashes }}"
       upload-assets: true # Optional: Upload to a new release
       provenance-name: kargo-cli.intoto.jsonl
+
+  update-homebrew-formula:
+    if: >
+      github.event_name == 'release' &&
+      startsWith(github.ref, 'refs/tags/v') &&
+      !contains(github.ref, '-') &&
+      github.ref == format('refs/tags/v{0}', steps.latest_release.outputs.version)
```

Comment:
I think you can do something like `github.event_name == 'release' && github.event.action == 'released'`, which would filter out prereleases. As our release candidates are typically marked as a pre-release, see e.g. https://github.com/akuity/kargo/releases/tag/v1.1.0-rc.3

Ref: https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#release

Reply:
The only thing my suggestion did not take into account is @krancour's

> [..] and patch releases not cut from the latest release-1.x branch

For this, I think we still need to pull off some version comparisons.

---

However, I also want to note that it's common for Homebrow to e.g. provide minor version taps. I.e., `kargo.rb` (latest), `kargo@1.1.rb` (1.1.x releases), `kargo@1.0.rb` (1.0.x releases), etc.

In the future, we should offer support for this as well. Without this, it would be impossible to update to a patch release for a previous minor version if it's released after a newer minor range.